projects/robots/boston_dynamics/spot/controllers/spot_moving_demo/spot_moving_demo.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 1996-2020 Cyberbotics Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Description:   Simple controller to present the Spot robot.
+ */
+
+#include <webots/camera.h>
+#include <webots/device.h>
+#include <webots/led.h>
+#include <webots/motor.h>
+#include <webots/robot.h>
+
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+#define NUMBER_OF_LEDS 8
+#define NUMBER_OF_JOINTS 12
+#define NUMBER_OF_CAMERAS 5
+
+// Initialize the robot's information
+static WbDeviceTag motors[NUMBER_OF_JOINTS];
+static const char *motor_names[NUMBER_OF_JOINTS] = {
+  "front left leg shoulder abduction motor",  "front left leg shoulder rotation motor",  "front left leg elbow motor",
+  "front right leg shoulder abduction motor", "front right leg shoulder rotation motor", "front right leg elbow motor",
+  "rear left leg shoulder abduction motor",   "rear left leg shoulder rotation motor",   "rear left leg elbow motor",
+  "rear right leg shoulder abduction motor",  "rear right leg shoulder rotation motor",  "rear right leg elbow motor"};
```

Comment:
I am wondering whether we should remove the "leg" keyword from the names here as it seems to be unnecessary to combine with shoulder or elbow, etc.